docs/it/week11/11-2.md

```diff
@@ -0,0 +1,559 @@
+
+<!-- $$
+L(x,y)=\sum_{ij}\frac{max(0,1-(x[y[j]]-x[i]))}{x.\text{size}(0)}
+$$
+
+This margin-base loss allows for different inputs to have variable amounts of targets. In this case you have several categories for which you want high scores and it sums the  hinge loss over all categories. For EBMs, this loss function pushes down on desired categories and pushes up on non-desired categories. -->
+
+### Perdita di hinge ad immersione - `nn.HingeEmbeddingLoss()`
+
+<!-- ### Hinge Embedding Loss - `nn.HingeEmbeddingLoss()` -->
+
+$$
+l_n =
+\left\{
+             \begin{array}{lr}
+             x_n, &\quad y_n=1,  \\
+             \max\{0,\Delta-x_n\}, &\quad y_n=-1  \\
+             \end{array}
+\right.
+$$
+
+La perdita di _Hinge_ ad immersione viene utilizzata per l'apprendimento semi-supervisionato (semi-supervised learning) attraverso una misura della somiglianza fra due input. Questa funziona "ravvicina" gli esempi simili e allontana gli esempi dissimili l'uno dall'altro. La variabile $y$ ci indica la direzione (ravvicinamento vs. allontanamento) del punteggio per una coppia di input. Utilizzando una perdita _hinge_, il punteggio è positivo se $y$ è uguale a 1 ed è uguale ad un margine $\Delta$ se $y$ è uguale a -1.
+
+<!-- $$
+\begin{equation}
+l_n =
+\left\{
+             \begin{array}{lr}
+             x_n, &\quad y_n=1,  \\
+             \max\{0,\Delta-x_n\}, &\quad y_n=-1  \\
+             \end{array}
+\right.
+\end{equation}
+$$
+
+Hinge embedding loss used for semi-supervised learning by measuring whether two inputs are similar or dissimilar. It pulls together things that are similar and pushes away things are dissimilar. The $y$ variable indicates whether the pair of scores need to go in a certain direction. Using a hinge loss, the score is positive if $y$ is 1 and some margin $\Delta$ if $y$ is -1. -->
+
+### Perdita di coseno ad immersione - `nn.CosineEmbeddingLoss()`
+
+<!-- ### Cosine Embedding Loss - `nn.CosineEmbeddingLoss()` -->
+
+$$
+l_n =
+  \left\{
+             \begin{array}{lr}
+             1-\cos(x_1,x_2), & \quad y=1,  \\
+             \max(0,\cos(x_1,x_2)-\text{margine}), & \quad y=-1
+             \end{array}
+   \right.
+$$
+
+Questa funzione di perdita misura la somiglianza fra due input, utilizzando la distanza del coseno di similitudine, e viene tipicamente utilizzata per apprendere immersioni non-lineari o per l'apprendimento semi-supervisionato.
+
+* Definito in maniera alternativa, 1 meno il coseno dell'angolo fra due vettori è praticamente la distanza Euclidea normalizzata.
+* Il vantaggio <!-- svantaggio? --> di questa funzione è che quando si hanno due vettori e li si vuole rendere molto distanti fra di loro, lo si può ottenere semplicemente rendendo i vettori molto lunghi. Ovviamente questo non è ottimale, perché non vogliamo aumentare la lunghezza dei vettori ma ruotarli nella direzione giusta. Per questo, si normalizzano i vettori e si calcola la distanza Euclidea fra di loro.
```

Comment:
L'originale dice "advantage"... ma da ciò che viene descritto sembrerebbe un disadvantage perché poi si dice "this is not optimal."

Frase di riferimento: "The advantage of this is that whenever you have two vectors and you want to make their distance as large as possible, it is very easy to make the network achieve this by make the vectors very long. Of course this is not optimal."